### PR TITLE
🌱 Move addon manager from addon-framework to ocm repo

### PR DESCRIPTION
**Pull request by zhujian7**
## Summary
Move addon manager from repo [addon-framework](https://github.com/open-cluster-management-io/addon-framework/tree/0056b148e0a0621c716f1ceffaa034e0f5caaaf0/pkg/manager) to ocm repo.

- [x] move go files
- [x] move integration test
- [ ] move e2e tests: a bit complicated and will be tracked in a follow-up PR

## Related issue(s)
https://github.com/open-cluster-management-io/ocm/issues/128

### CODE REVIEW

File: pkg/addon/controllers/addonmanagement/controller.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+				addonClient:                addonClient,
+				placementDecisionLister:    placementDecisionInformer.Lister(),
+				placementLister:            placementInformer.Lister(),
+				managedClusterAddonIndexer: addonInformers.Informer().GetIndexer(),
+				addonFilterFunc:            addonFilterFunc,
+			},
+		},
+	}
+
+	return factory.New().WithInformersQueueKeysFunc(
```
Comment (qiujian16) [inline pkg/addon/controllers/addonmanagement/controller.go:65]: I think we can use library-go factory for all these controllers

**Comment by qiujian16** [comment]
@zhujian7  also need to stop the gh action on addon-framework repo at first.

**Comment by zhujian7** [comment]
@qiujian16 PTAL

**Pull request merged**